src/sentry/tasks/deletion.py

```diff
@@ -308,6 +311,42 @@ def delete_tag_key(object_id, transaction_id=None, continuous=True, **kwargs):
     })
 
 
+@instrumented_task(name='sentry.tasks.deletion.generic_delete', queue='cleanup',
+                   default_retry_delay=60 * 5, max_retries=None)
+@retry(exclude=(DeleteAborted,))
+def generic_delete(app_label, model_name, object_id, transaction_id=None,
+                   continuous=True, actor_id=None, **kwargs):
+    from sentry.models import User
+
+    model = get_model(app_label, model_name)
+
+    try:
+        instance = model.objects.get(id=object_id)
+    except model.DoesNotExist:
+        return
+
+    if instance.status == ObjectStatus.VISIBLE:
+        raise DeleteAborted
+
+    if instance.status == ObjectStatus.PENDING_DELETE:
+        if actor_id:
+            actor = User.objects.get(id=actor_id)
```

Comment:
In the case of this user_id not existing. If the user account were deleted or something, this will cause the deletion job to never finish since this will error and just keep retrying.